src/targets/node/fetch.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @description
+ * HTTP code snippet generator for Node.js using node-fetch.
+ *
+ * @author
+ * @hirenoble
+ *
+ * for any questions or issues regarding the generated code snippet, please open an issue mentioning the author.
+ */
+
+'use strict'
+
+var stringifyObject = require('stringify-object')
+var CodeBuilder = require('../../helpers/code-builder')
+
+module.exports = function (source, options) {
+  var opts = Object.assign({
+    indent: '  '
+  }, options)
+
+  var includeFS = false
+  var code = new CodeBuilder(opts.indent)
+
+  code.push('const fetch = require(\'node-fetch\');')
+  var url = source.url
+  var reqOpts = {
+    method: source.method
+  }
+
+  if (Object.keys(source.queryObj).length) {
+    reqOpts.qs = source.queryObj
+  }
+
+  if (Object.keys(source.headersObj).length) {
+    reqOpts.headers = source.headersObj
+  }
+
+  switch (source.postData.mimeType) {
+    case 'application/x-www-form-urlencoded':
+      reqOpts.body = source.postData.paramsObj
+      break
+
+    case 'application/json':
+      if (source.postData.jsonObj) {
+        reqOpts.body = source.postData.jsonObj
+        reqOpts.json = true
```

Comment:
Interesting, does this need to also be encoded to JSON body, or does the `reqOpts.json = true` handle it? I couldn't find docs on `node-fetch` about the json option does, but looking at the example, it looks as though we _do_ need to encode: https://github.com/node-fetch/node-fetch#post-with-json.

Reply:
Good catch. Honestly I didn't take much attention on the existing code...

Reply:
I hadn't reviewed it previously, so took the chance to now. Rest looks good though I think!